scripts/migrate-users.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import argparse
+from datetime import datetime
+import sys
+import json
+import yaml
+import urllib
+import requests
+import time
+
+# import logging
+# import httplib as http_client
+# http_client.HTTPConnection.debuglevel = 1
+# logging.basicConfig()
+# logging.getLogger().setLevel(logging.DEBUG)
+# requests_log = logging.getLogger("requests.packages.urllib3")
+# requests_log.setLevel(logging.DEBUG)
+# requests_log.propagate = True
+
+# If you're running into M_NOT_JSON issues, python-requests strips the body of request on 301 redirects. Make sure you're using the direct url of your homeserver.
+# See https://github.com/requests/requests/issues/2590
+
+def get_appservice_token(reg):
+    with open(reg, "r") as f:
+        reg_yaml = yaml.load(f)
+        return reg_yaml["as_token"]
+
+def get_users(homeserver, room_id, token, user_prefix, name_suffix):
+    res = requests.get(homeserver + "/_matrix/client/r0/rooms/" + urllib.quote(room_id) + "/joined_members?access_token=" + token)
+    joined = res.json().get("joined", None)
+    user_ids = [user_id for user_id in joined if user_id.startswith(user_prefix) and (joined.get(user_id).get("display_name") or "").endswith(name_suffix) ]
+    return { uid: joined.get(uid).get("display_name") for uid in user_ids }
+
+def get_rooms(homeserver, token):
+    res = requests.get(homeserver + "/_matrix/client/r0/sync?access_token=" + token).json()
```

Comment:
You don't want to do this. On large bridges this will absolutely destroy Synapse. Use `/joined_rooms` instead. https://github.com/matrix-org/matrix-appservice-bridge/blob/8f60f3e84a98d8bfd4dbd52d5403ff1c841f7d32/lib/components/app-service-bot.js#L40